lib/autoyast.pm

```diff
@@ -783,6 +784,73 @@ sub expand_agama_profile {
     return $profile_url;
 }
 
+=head2 workaround_install_jsonnet
+
+ workaround_install_jsonnet();
+
+ Workaround Install via zypper golang-github-google-jsonnet package in the worker.
+ We need this meanwhile the package is built for all other architectures (s390x/aarch64/ppc64le).
+ After that it should be in the Salt repository.
+
+=cut
+
+sub workaround_install_jsonnet {
+    if (system("sudo", "zypper", "-n", "repos", "systemsmanagement_Agama_Devel") != 0) {
+        system("sudo", "zypper", "-n", "addrepo", "-f", "-G",
+            "https://download.opensuse.org/repositories/systemsmanagement:Agama:Devel/15.6/systemsmanagement:Agama:Devel.repo");
+    }
+
+    if (system("command", "-v", "jsonnet") != 0) {
+        system("sudo", "zypper", "-n", "install", "-f", "golang-github-google-jsonnet");
+    }
+}
+
+=head2 parse_profile_options
+
+ parse_profile_options(options => $options);
+
+ Parse given options to match required parameters syntax of jsonnet.
+
+=cut
+
+sub parse_profile_options {
+    my ($options) = @_;
+
+    my @params = split(' ', trim($options));
+    for my $element (@params) {
+        if ($element =~ /true/ || $element =~/false/) {
+            $element = " --tla-code " . $element
+        } else {
+            $element = " --tla-str " . $element
+        }
+    }
+    return @params
+}
+
+=head2 generate_json_profile
+
+ generate_json_profile();
+
+ Return the URL of generated JSON profile
+
+=cut
+
+sub generate_json_profile {
+    my $profile_name = "generated_profile.json";
+    my $profile_options = parse_profile_options(get_var('AGAMA_PROFILE_OPTIONS'));
```

Comment:
the lines which contains some logic should go below, the casedir is just an assignation and should go here instead.